signer/gpg2/gpg2.go

```diff
@@ -269,3 +354,86 @@ type Options struct {
 func (s *GPG2Signer) GetDefaultOptions() interface{} {
 	return Options{}
 }
+
+// SignFiles uses debsign to gpg2 clearsign multiple named
+// *.buildinfo, *.dsc, or *.changes files
+func (s *GPG2Signer) SignFiles(inputs []signer.NamedUnsignedFile, options interface{}) (signedFiles []signer.NamedSignedFile, err error) {
+	if s.Mode != ModeDebsign {
+		err = fmt.Errorf("gpg2: can only sign multiple files in %s mode", ModeDebsign)
+		return
+	}
+
+	// create a tmp dir outside the signer GPG home
+	inputsTmpDir, err := ioutil.TempDir("", fmt.Sprintf("autograph_%s_%s_%s_sign_files", s.Type, s.KeyID, s.Mode))
+	if err != nil {
+		err = fmt.Errorf("gpg2: error creating tempdir for debsign: %w", err)
+		return
+	}
+	defer os.RemoveAll(inputsTmpDir)
```

Comment:
Hmm yeah good call! I thought `defer` would propagate errors somehow.

I think it's OK if we don't clean up the input files, but we definitely want to know if we're failing to delete them.

So:

142025daae888f1f43251e2b43e51bc1436c6769 / 4683927 will warn but not return an error when input files or SignData content file cleanup fails
b8962b7fb579695d79f3ba4ce88c341a8d5ce165 will error if pub and private key cleanup fails, since leaving them on disk is an error. Other errors (e.g. key import failure) will take precedence.